app/callbacks.py

```diff
@@ -68,17 +70,44 @@ def disable_tabs(file_name):  # noqa: D103
 
 # Define another callback to access the stored file path and read the file
 @app.callback(
+    Output("gm-graph", "figure"),
+    Output("gm-graph", "style"),
     Output("file-content-mg", "children"),
     [Input("file-store", "data")],
 )
-def display_file_contents(file_path):  # noqa: D103
+def gm_plot(file_path):  # noqa: D103
     if file_path is not None:
         with open(file_path, "rb") as f:
             data = pickle.load(f)
         # Process and display the data as needed
-        content = f"File contents: {data[0][:2]}"
-        return content  # Display same content in both tabs
-    return "No data available"
+        _, gcfs, _, _, _, _ = data
+        n_bgcs = {}
+        for gcf in gcfs:
+            n_bgcs[gcf.id] = len(gcf.bgcs)
+
+        keys_by_value = defaultdict(list)
+
+        for key, value in n_bgcs.items():
+            keys_by_value[value].append(key)
```

Comment:
```suggestion
        n_bgcs = {}
        for gcf in gcfs:
            n = len(gcf.bgcs)
            if n not in n_bgcs:
                n_bgcs[n] = [gcf.id]
            else:
                n_bgcs[n].append(gcf.id)
```

The `defaultdict` is advanced and could be replaced by more friendly `if-else`.